extensions/objects/templates/NumberWithUnitsEditor.js

```diff
@@ -0,0 +1,53 @@
+// Copyright 2018 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+oppia.directive('numberWithUnitsEditor', [
+  '$compile', 'NumberWithUnitsObjectFactory', 'OBJECT_EDITOR_URL_PREFIX',
+  function($compile, NumberWithUnitsObjectFactory, OBJECT_EDITOR_URL_PREFIX) {
+    return {
+      link: function(scope, element) {
+        scope.getTemplateUrl = function() {
+          return OBJECT_EDITOR_URL_PREFIX + 'NumberWithUnits';
+        };
+        $compile(element.contents())(scope);
+      },
+      restrict: 'E',
+      scope: true,
+      template: '<span ng-include="getTemplateUrl()"></span>',
+      controller: ['$scope', function($scope) {
+        var errorMessage = '';
+        if ($scope.$parent.value !== null) {
+          $scope.localValue = {
+            label: defaultFractionString
```

Comment:
Where is this defined? Also, it should be defaultNumberWithUnitsString when `$scope.$parent.value === null`. When it is not null you have to convert it to dict and then back to string and then show it.

Reply:
Done.